.circleci/config.yml

```diff
@@ -9,7 +9,87 @@ defaults: &defaults
     GOPATH: /go
     SKIP_CLEANUP: true
 
+# VM environment. Includes docker.
+integrationDefaults: &integrationDefaults
+  machine: true
+  working_directory: /go/src/istio.io/cni #same as docker
+  environment:
+    - CHANGE_MINIKUBE_NONE_USER: true
+    - GOPATH: /go
+    - SKIP_CLEANUP: true
+    - KUBECONFIG: /go/out/minikube.conf
+    - TEST_ENV: minikube-none
+
+# Common procedure to initialize working directory
+initWorkingDir: &initWorkingDir
+  type: shell
+  name: Initialize Working Directory
+  pwd: /
+  command: |
+    sudo mkdir -p /go/src/istio.io/cni
+    sudo chown -R circleci /go
+    sudo mkdir -p /go/src/istio.io/istio
+    mkdir -p /go/out/tests
+    mkdir -p /go/out/logs
+    mkdir -p /home/circleci/logs
+
+recordZeroExitCodeIfTestPassed: &recordZeroExitCodeIfTestPassed
+  run:
+    when: on_success
+    name: Record zero exit code as test passed
+    command: echo 0 > exit_code
+
+recordNonzeroExitCodeIfTestFailed: &recordNonzeroExitCodeIfTestFailed
+  run:
+    when: on_fail
+    name: Record nonzero exit code as test failed
+    command: echo 1 > exit_code
+
+markJobStartsOnGCS: &markJobStartsOnGCS
+  run:
+    when: always
+    command: bin/ci2gubernator.sh --job_starts
+
+markJobFinishesOnGCS: &markJobFinishesOnGCS
+  run:
+    when: always
+    command: |
+      make dumpsys || true
+      make junit-report || true
+      # TODO: upload the artifacts as well, for debugging !
+      bin/ci2gubernator.sh --exit_code=$(cat exit_code) --junit_xml=/go/out/tests/junit.xml
+
 jobs:
+  e2e-simple-cni:
+    <<: *integrationDefaults
+    environment:
+            - KUBECONFIG: /go/out/minikube.conf
+            - TEST_ENV: minikube-none
+            - GOPATH: /go
+    steps:
+      - <<: *initWorkingDir
+      - checkout
+      - attach_workspace:
+          at:  /go
+      - <<: *markJobStartsOnGCS
+      - run: make sync
+      - run: bin/testEnvRootMinikube.sh start
+      - run: make build
+      - run: bin/testEnvRootMinikube.sh wait
+      - run: docker images
+      - run: make dummy_test
+      - <<: *recordZeroExitCodeIfTestPassed
+      - <<: *recordNonzeroExitCodeIfTestFailed
+      - <<: *markJobFinishesOnGCS
+      - store_artifacts:
+          path: /home/circleci/logs
+      - store_artifacts:
+          path: /tmp
+      - store_artifacts:
+          path: /var/lib/localkube/
+      - store_test_results:
+          path: /go/out/tests
+ 
```

Comment:
need to reference this job from the workflows section to get it to run.  The "all" list runs on PRs.